src/lib/accessibility/FocusZone/FocusZone.tsx

```diff
@@ -290,6 +295,8 @@ export class FocusZone extends React.Component<IFocusZoneProps> implements IFocu
     if (stopFocusPropagation) {
       ev.stopPropagation()
     }
+
+    _.invoke(this.props, 'onFocus', ev)
```

Comment:
Good point. I did this because I changed the meaning of this callback to conform to unit tests. The event should now behave as a transparent `onFocus` callback, i.e., you can do `<FocusZone onFocus={...}>...</FocusZone>` and in that case, I personally expect the callback to be called *after* the component itself handles the callback, so the order is:

1. the component itself handles the callback,
2. external callbacks are called.

We can discuss this.

Reply:
Makes sense. I just would like to be consistent. Currently, `Portal`, `Input`, maybe others call `onXXX()` before `trySetState`, but that's out of scope of this PR.